libs/designsystem/src/lib/components/list/list.component.ts

```diff
@@ -238,4 +240,8 @@ export class ListComponent implements OnInit, AfterViewInit, OnChanges {
       this.isSwipingEnabled = this.platform.isTouch();
     }
   }
+
+  public enableLoadOnDemand(isLoadOnDemandEnabled: boolean) {
+    this.isLoadOnDemandEnabled = isLoadOnDemandEnabled;
+  }
```

Comment:
Couldn't we just make the existing property `isLoadOnDemandEnabled` an `@Input` property?
Why the need for a method?

Reply:
With my current understanding of Angular; making `isLoadOnDemandEnabled` an `@input` property would move the responsibility of determining the state of `isLoadOnDemandEnabled` to the parent component that is utilising the `list` component. But the state can currently be altered by the `done` function exposed by the `loadOnDemand` event. 

Marking `isLoadOnDemandEnabled` with `@input` would result In there being two ways the value can be altered: with the input property and via the `done` function.  

The parent component will not be notified if the state of `isLoadOnDemandEnabled` is changed via `done` (unless an event emitter or callback is implemented). To my understanding, this will result in `isLoadOnDemand` being perceived as having two different values: 
   1. The parent component trying to control `isLoadOnDemandEnabled` via an input property will have the initial value. 
   2. The `list` component with the `isLoadOnDemandEnabled` member will have the altered value that is set by `done`. 

Now that I think about it - my understanding of how internal changes to input properties are handled might be wrong. But even if an internal change to an input property is reflected in the parent, it would mean we're breaking the unidirectional dataflow.

 My tought was that It was simpler to keep the responsibility of controlling the state of `isLoadOnDemand` in the `list`-component, as we're already using `done` to control the state of `isLoadOnDemandEnabled`. The component can then be "told" what we want with `enableLoadOnDemand`, moving the responsibility of setting the value back to the component itself. 

My own preferred solution would be to remove the `done` function and then only being able to control the state of `isLoadOnDemandEnabled` via an input property as this to me is cleaner. But I suppose there's a reason why it is done via an function on an event and altering it might be a breaking change 😄

Reply:
That might be true under other circumstances, like encapsulating the state management to the component, like having readonly props and some checks when trying to alter state through a method.

But in this particular case, it's still the consumer that sets the state - through a method, but tomato/tomato - and there are no checks/guards, side effects or anything going on... So it's kinda redundant to wrap that property in a method (and by the way: The property is already public, so anyone can still set it ;) )

In an Angular context ( or any declarative context really), by exposing the property it's possible to do this:
`<kirby-list [isLoadOnDemandEnabled]="someAsyncOperationReturningBoolean | async">`

With a method, we would have to expose the list instance through a query and call the method on that instance when the value changes. It's doable, but not as straight forward.

Reply:
I see your point and will revert to using an input property instead. I've got some follow up questions which I might bother you with Monday 😄